bot/message_processor.go

```diff
@@ -2,116 +2,137 @@ package bot
+		switch {
+		case bot.isMasterUser(user):
+			{
+				bot.masterMessages(message)
+			}
+		default:
+			{
+				bot.niceMessage(message, user)
+			}
 		}
-	// If the user is an offending user message them, but remove them off the list
-	case isViolator:
-		{
-			bot.violatorMessage(message, user)
-		}
-		// If this is a message directed at the bot respond in a nice way
-	case strings.HasPrefix(message.Text, fmt.Sprintf("<@%s>", bot.Slack.ID)):
-		{
-			bot.niceMessage(message, user)
+	} else {
+		switch {
+		// Messages that contain the word tick
+		case strings.Contains(message.Text, " tick "):
+			{
+				bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(
+					"tock", message.Channel,
+				))
+			}
+		// Messages that references the bot will be send out 30% of the time. See: Foucault, Discipline 201
+		case strings.Contains(message.Text, fmt.Sprintf("<@%s>", bot.Slack.GetSelfID())):
+			{
+				var returnMessage string
+				randomInt := rand.Intn(100)
+				if randomInt >= 70 {
+					returnMessage = bot.MessageRepo.Nice.GenerateMessage(user)
+				} else if randomInt <= 3 {
+					returnMessage = "'The [tockers] must never know whether [they are] looked at at any one moment;"
+					returnMessage += " but [they] must be sure that [they] may always be so' - Foucault, Discipline 201"
+				}
+				bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(
+					returnMessage,
+					message.Channel,
+				))
+			}
 		}
 	}
 }
 
+// violatorMessage has the message for a late user
+func (bot *Bot) violatorMessage(message *slack.MessageEvent, user string) {
+	var returnMessage string
+	// Check if user is still late
+	if bot.isLateUser(user) {
+		returnMessage = bot.MessageRepo.Angry.GenerateMessage(user)
+	} else {
+		returnMessage = fmt.Sprintf(
+			"<@%s>, I was about to yell at you, but then I realized you actually filled out your timesheet. Thanks! ^_^",
+			user,
+		)
+	}
+	bot.violatorUserMap.Delete(user)
+	bot.Slack.SendMessage(bot.Slack.NewOutgoingMessage(returnMessage, message.Channel))
+}
+
 // masterMessages contains the commands for admins
-func (bot *Bot) masterMessages(message slackPackage.Message) {
+func (bot *Bot) masterMessages(message *slack.MessageEvent) {
+	var returnMessage string
 	switch {
 	case strings.Contains(message.Text, "slap users"):
 		{
 			go bot.SlapLateUsers()
-			message.Text = "Slapping Users!"
+			returnMessage = "Slapping Users!"
 		}
 	case strings.Contains(message.Text, "bother users"):
 		{
 			bot.startviolatorUserMapUpdater()
-			message.Text = "Starting to bother users!"
+			returnMessage = "Starting to bother users!"
 		}
 	case strings.Contains(message.Text, "who is late?"):
 		{
 			lateList, total := bot.fetchLateUsers()
-			message.Text = fmt.Sprintf("%s are late! %d people total.", lateList, total)
+			returnMessage = fmt.Sprintf("%s are late! %d people total.", lateList, total)
 		}
 	default:
 		{
-			message.Text = fmt.Sprintf(
+			returnMessage = fmt.Sprintf(
```

Comment:
create a variable called `id` or something to prevent having to call the method multiple times.